logstash-core/lib/logstash/agent_plugin_manager.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+require "logstash/agent"
+
+module LogStash module AgentPluginManager
+  DEFAULT_AGENT_NAME = :default
+  class DuplicatePluginNameError < Error; end
+
+  REGISTRY = {}
+
+  # Reset plugin registry to just the default plugin
+  def self.reset!
+    REGISTRY.clear
+    REGISTRY[DEFAULT_AGENT_NAME] = LogStash::Agent
+  end
+
+  reset!
```

Comment:
@purbon well, where would you put this logic? We need to set this somewhere, and this is a default property of this module.

The only alternative I can think of would be rewriting the module as a class and putting it in the initializer. But that seems like excessive ceremony to work around something that could be implemented simply.

I stand by my assertion that an explanatory comment is all that's needed here. Maybe @jordansissel can chime in with a tie breaking opinion here.